### DISCUSSION TITLE
Feature request: public mode *with funding offer*

### DISCUSSION THREAD

**Discussion by actuallymentor** [wg-easy/wg-easy#1055] [Ideas]
Hello fellow countryman @WeeJeWel!

I have a featureset request and am willing to pay to have it implemented (and keep it open source under an MIT license). Please let me know if this is something you or another maintainer is open to.

Usecase: I want to deploy wg-easy and let an arbitraty amount of unauthenticated users request WG config files. In essence to run public-service servers. I understand the risks and intricacies involved (I currently run generousvpn.com with a similar model using openvpn).

I am willing to pay you or other maintainers to implement the following feature set.

Payable tier 1:

- [ ] `DANGEROUS_PUBLIC_MODE=true` environment variable that allows unauthenticated users to request wg configs through the web interface. This only requires inputting an email address (as a contactable user id)
- [ ] `AUTO_ACCEPT_PUBLIC_REQUESTS=true` behaviour that makes the public mode behave like the admin mode does now (ie offers user to download configs, note: this is isolated so users do not see eachother's configs)
- [ ] When `DANGEROUS_PUBLIC_MODE` is enabled, the wireguard defaults isolate users and allow for arbitrary amounts of user signups

Payable tier 2:

- [ ] `AUTO_ACCEPT_PUBLIC_REQUESTS=false` behaviour that ads a "moderation" queue to the admin panel that allows an admin to accept or deny request

Payable tier 3:

- [ ] `EXPIRE_PUBLIC_CONFIGS=amount_in_days` variable that deletes configs (and releases that client ip) after x days of inactivity from the client
- [ ] `WEBHOOK_URL` environment variable that triggers webhook calls on public request creation

**Comment by actuallymentor** [comment]
Afternoon all! I see some labeling activity, should I read that as potential interest in pursuing this? I'm happy to discuss in here or if you prefer a non-public channel that is ok too. I see @pheiduck looks the most active, would you be interested?

**Reply by pheiduck** [reply]
I would welcome this. However, not in the foreseeable future say that this will become a paid part-time job.

**Comment by pheiduck** [comment]
I think we should discuse this in a none-public channel. We should also wait of a response from @WeeJeWel and @peterlewis.

**Comment by actuallymentor** [comment]
I'm available on Signal/WhatsApp and email on `+31631250266` and `mentor@palokaj.co`
